pkg/broker/ingress/handler.go

```diff
@@ -55,21 +61,21 @@ type HttpMessageReceiver interface {
 }
 
 // handler receives events and persists them to storage (pubsub).
-// TODO(liu-cong) add metrics
-// TODO(liu-cong) add tracing
 // TODO(liu-cong) support event TTL
 type handler struct {
 	// httpReceiver is an HTTP server to receive events.
 	httpReceiver HttpMessageReceiver
 	// decouple is the client to send events to a decouple sink.
 	decouple DecoupleSink
 	logger   *zap.Logger
+	ctx      context.Context
```

Comment:
Context should always use the current request context and should not be stored in a struct. I recommend applying the tag mutators to the incoming request context. This has the advantage preserving any incoming tags.

Reply:
The ctx is used to store the common pod and container name tags. Brought back StatsReporter to hold the pod and container names.